src/resources/js/tribe-events-bar.js

```diff
@@ -140,78 +142,166 @@ var tribe_events_bar_action;
-			} ).html( [
-				'   <a href="#">',
-				'   <span class="tribe-icon-' + displaying + '">' + $view.text() + '</span>',
-				'</a>'].join( "" )
-			).appendTo( '.tribe-bar-views-list' );
+				'data-view'           : displaying,
+				'aria-labelledby'     : 'tribe-bar-views-label ' + unique_c,
+			} ).html( '<span class="tribe-icon-' + displaying + '" aria-hidden="true" role="none"></span>' + $view.text() ).appendTo( '.tribe-bar-views-list' );
 
 		} );
 
 		//find the current view and select it in the bar
-		var currentview = $tribebarselect.find( ':selected' ).data( 'view' );
-		var $currentli  = $tribebarviews.find( 'li[data-view=' + currentview + ']' );
+		var $currentli = getCurrentViewItem();
+
+		// Se a class on the current view element
+		$currentli.addClass( 'tribe-bar-active' );
+
+		// Create the listbox toggle button
+		var $tribebarviewstoggle = $( '<button>', {
+			'id'              : 'tribe-bar-views-toggle',
+			'class'           : 'tribe-bar-views-toggle',
+			'data-view'       : $currentli.data( 'view' ),
+			'aria-haspopup'   : 'listbox',
+			'aria-labelledby' : 'tribe-bar-views-label tribe-bar-views-toggle',
+		} );
 
-		$currentli.prependTo( $tribebarviews ).addClass( 'tribe-bar-active' );
+		$tribebarviewstoggle.html( $currentli.html() ).insertBefore( $tribebarviews );
 
 		// Disable the select
 		$tribebarselect.hide();
 
+		function openViewsToggle() {
+			var $currentli = getCurrentViewItem();
+			$tribebarviews
+				.slideDown( 'fast' )
+				.attr( 'aria-activedescendant', $currentli.attr( 'id' ) )
+				.focus();
+			$tribebar.addClass( 'tribe-bar-views-open' );
+			$tribebarviewstoggle.attr( 'aria-expanded', 'true' );
+		}
+
+		function closeViewsToggle() {
+			var $currentli = getCurrentViewItem();
+			$tribebarviewstoggle.removeAttr( 'aria-expanded' );
+			$tribebar.removeClass('tribe-bar-views-open');
+			$tribebarviews
+				.slideUp( 'fast' )
+				.removeAttr( 'aria-activedescendant' )
+				.find( '.tribe-bar-views-option' ).removeClass( 'tribe-bar-active' );
+			$currentli.addClass( 'tribe-bar-active' );
+		}
+
+		function triggerViewsChange( e, el ) {
+			e.preventDefault();
+
+			var $this = $( el );
+
+			// If the new selection is the same as the current view, just close the drop-down and bail.
+			if ( $this.data( 'view' ) === $tribebarviewstoggle.data( 'view' ) ) {
+				$tribebarviewstoggle.focus();
+				closeViewsToggle();
+				return;
+			}
+
+			// Otherwise, update the page with the selected view and trigger the change
+			$tribebarviewstoggle.html( $this.html() ).focus();
+			closeViewsToggle();
+
+			ts.cur_url              = $( 'option[data-view=' + $this.data( 'view' ) + ']' ).val();
+			ts.view_target          = $( 'select[name=tribe-bar-view] option[value="' + ts.cur_url + '"]' ).data( 'view' );
+			tribe_events_bar_action = 'change_view';
+
+			tribe_events_bar_change_view();
+		}
+
 		// toggle the views dropdown
-		$tribebar.on( 'click', '#tribe-bar-views', function( e ) {
```

Comment:
Are we updating/changing the ID of this element? Let's be sure we document it and communicate in the release notes in case somebody styled this element in their tribe bar. Users can get really angry sometimes.

Reply:
No change the element's ID, no. I've added a new button element that is used for triggering the views drop-down toggle (instead of toggling on the whole #tribe-bar-views container div), but the views container element is keeping its existing ID. I tied very hard not to change any existing IDs/classes. I only added new elements, classes, IDs as necessary to resolve the a11y issues.

Reply:
Awesome, thanks Dave :)